benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,736 @@
+            If True, the graph will be renumbered.
+    Returns:
+        G:  cugraph.Graph
+    """
+    assert dask_dataframe['src'].dtype == 'int64'
+    assert dask_dataframe['dst'].dtype == 'int64'
+
+    if 'etp' in dask_dataframe.columns:
+        assert dask_dataframe['etp'].dtype == 'int32'
+
+    G = cugraph.MultiGraph(directed=True)
+    G.from_dask_cudf_edgelist(
+        dask_dataframe,
+        source="src", 
+        destination="dst",
+        edge_type='etp' if 'etp' in dask_dataframe.columns else None,
+        renumber=False
+    )
+    return G
+
+
+def symmetrize_ddf(dask_dataframe):
+    source_col, dest_col = symmetrize(
+        dask_dataframe,
+        'src',
+        'dst',
+        multi=True,
+        symmetrize=True,
+    )
+
+    new_ddf = source_col.to_frame()
+    new_ddf['dst'] = dest_col
+
+    return new_ddf
+
+def renumber_ddf(dask_df, persist=False):
+    vertices = dask_cudf.concat([dask_df['src'], dask_df['dst']]).unique().reset_index(drop=True)
+    if persist:
+        vertices = vertices.persist()
+    
+    vertices.name = 'v'
+    vertices = vertices.reset_index().set_index('v').rename(columns={'index': 'm'})
+    if persist:
+        vertices = vertices.persist()
+
+    src = dask_df.merge(vertices, left_on='src', right_on='v', how='left').m.rename('src')
+    dst = dask_df.merge(vertices, left_on='dst', right_on='v', how='left').m.rename('dst')
+    df = src.to_frame()
+    df['dst'] = dst
+
+    return df.reset_index(drop=True)
+
+def _make_batch_ids(bdf: cudf.DataFrame, batch_size: int, num_workers: int, partition_info: Optional[Union[dict, str]] = None):
+    # Required by dask; need to skip dummy partitions.
+    if partition_info is None:
+        return cudf.DataFrame({
+            'batch': cudf.Series(dtype='int32'),
+            'start': cudf.Series(dtype='int64')
+        })
+    
+    partition = partition_info['number']
+    if partition is None:
+        raise ValueError('division is absent')
+
+    num_batches = int(ceil(len(bdf) / batch_size))
+    
+    batch_ids = cupy.repeat(
+        cupy.arange(num_batches * partition, num_batches * (partition + 1), dtype='int32'),
+        batch_size
+    )[:len(bdf)]
+
+    bdf = bdf.reset_index(drop=True)
+    bdf['batch'] = cudf.Series(batch_ids)
+
+    return bdf
+
+
+def _replicate_df(df: cudf.DataFrame, replication_factor: int, col_offsets:Dict[str, int], partition_info: Optional[Union[dict, str]] = None):
+    # Required by dask; need to skip dummy partitions.
+    if partition_info is None:
+        return cudf.DataFrame({
+            'batch': cudf.Series(dtype='int32'),
+            'start': cudf.Series(dtype='int64')
+        })
+    
+    if replication_factor > 1:
+        for r in range(1, replication_factor):
+            df_replicated = df
+            for col, offset in col_offsets.items():
+                df_replicated[col] += offset * r
+        
+            df = cudf.concat([df, df_replicated], ignore_index=True)
```

Comment:
For a factor of 4 we get 8x edges is this  expected ? I would have imagined  we would only have 4x edges. 
 
 ```python3
In [1]: import pandas as pd

In [2]: df = pd.DataFrame({"src":[0],"dst":[1]})

In [3]: replication_factor = 4

In [4]: col_offsets = {"src":0, "dst":0}

In [5]: for r in range(1, replication_factor):
   ...:     df_replicated = df
   ...:     for col, offset in col_offsets.items():
   ...:         df_replicated[col] += offset * r
   ...:     df = pd.concat([df, df_replicated], ignore_index=True)
   ...: 

In [6]: print(df)
   src  dst
0    0    1
1    0    1
2    0    1
3    0    1
4    0    1
5    0    1
6    0    1
7    0    1

 ```
```suggestion
        df_replicated = df
        for r in range(1, replication_factor):
            for col, offset in col_offsets.items():
                df_replicated[col] += offset * r
            df = cudf.concat([df, df_replicated], ignore_index=True)
```

Reply:
Should be fixed now